common/anything-sync-daemon.in

```diff
@@ -1,287 +1,520 @@
-
-debug "${RED}ASDNAME:${NRM} $ASDNAME"
-debug "${BLU}Configuration file:${NRM} $ASDCONF"
-debug "${GRN}Daemon file:${NRM} $DAEMON_FILE"
-debug "${RED}Lockfile:${NRM} $LOCKFILE\n"
-
-# Setup check /etc/asd.conf
-debug "Checking for existence of $ASDCONF"
-if [[ -f "$ASDCONF" ]]; then
-debug "Checking if asd is already running"
-  if [[ ! -f "$DAEMON_FILE" ]]; then
-    debug "Daemon file not found: asd is not running"
-    # do nothing if asd is currently running, otherwise
-    # make sure only comments and variables/arrays are defined to prevent
-    # problems
-    if grep -Eqv "^$|^\s*['\")#]|^[^ ]*=[^;]*" "$ASDCONF"; then
-      # found text that isn't a blank line, comment, or variable present so exit
-      echo -e " ${RED}ERROR:${NRM}${BLD} Syntax error(s) detected in ${BLU}$ASDCONF${NRM}"
-      echo -e "${NRM}${BLD}Line number: offending comment${NRM}"
-      grep -Evn "^$|^\s*['\")#]|^[^ ]*=[^;]*" "$ASDCONF"
-      exit 1
+  done
+}
+
+running_as_root() {
+  local euid="${EUID:-$(id -u)}" || return
+  (( euid == 0 ))
+}
+
+run_systemctl_system() {
+  systemctl --system "$@"
+}
+
+run_systemctl_user() {
+  systemctl --user "$@"
+}
+
+systemd_unit_name() {
+  [[ -v INVOCATION_ID ]] || return
+
+  local id invocation_id
+  while read -r prop; do
+    case "$prop" in
+      InvocationID=*)
+        invocation_id="${prop#InvocationID=}"
+        ;;
+      Id=*)
+        id="${prop#Id=}"
+        ;;
+      *)
+        if [[ -v id ]] && [[ -v invocation_id ]] && [[ "$invocation_id" == "$INVOCATION_ID" ]]; then
+          printf -- '%s' "$id"
+          return
+        else
+          unset id invocation_id
+        fi
+        ;;
+    esac
+  done < <(run_systemctl show -p 'Id,InvocationID' '*')
+
+  return 1
+}
+
+looks_like_systemd_system() {
+  systemd_unit_name &>/dev/null \
+    || run_systemctl_system list-unit-files asd.service &>/dev/null \
+    || [[ -f /usr/lib/systemd/system/asd.service ]]
+}
+
+looks_like_systemd_user() {
+  systemd_unit_name &>/dev/null \
+    || run_systemctl_user list-unit-files asd.service &>/dev/null \
+    || [[ -f /usr/lib/systemd/user/asd.service ]]
+}
+
+is_truthy() {
+  local b="${1?}"
+  case "${b,,}" in
+    y|yes|true|t|on|1|enabled|enable|use)
+      return 0
+      ;;
+    *)
+      return 1
+      ;;
+  esac
+}
+
+serialize_conf_vars() {
+  # These variables are meant to be declared in `asd.conf`, and we gracefully
+  # handle them being unset.
+  # shellcheck disable=SC2153
+  declare -p \
+    BACKUP_LIMIT  \
+    DEBUG \
+    ENABLE_HARDLINK_SAFETY_CHECK \
+    USE_BACKUPS \
+    USE_OVERLAYFS \
+    VOLATILE \
+    WHATTOSYNC 2>/dev/null || :
+}
+
+serialize_conf_vars_with_timeout() {
+  export -f serialize_conf_vars
+
+  # shellcheck disable=SC2016
+  timeout "${ASDCONFTIMEOUT:-10}" "$BASH" "-$-" -c '
+    source "${1?}" || exit
+    serialize_conf_vars
+  ' "$0" "${1?}"
+}
+
+use_v1_paths() {
```

Comment:
I am not sure about why we are changing paths btw, what benefits do we get and what paths are we ending up changing?